gw_eccentricity/eccDefinition.py

```diff
@@ -1350,26 +1442,85 @@ def measure_ecc(self, tref_in=None, fref_in=None):
+        # frequency where these are measured.
+        return self.make_return_dict_for_eccentricity_and_mean_anomaly()
+
+    def set_eccentricity_and_mean_anomaly_to_zero(self):
+        """Set eccentricity and mean_anomaly to zero."""
+        if self.domain == "time":
+            # This function sets eccentricity and mean anomaly to zero when a
+            # method fails to detect any extrema, and therefore, in such cases,
+            # we can set tref_out to be the times that fall within the range of
+            # self.t.
+            self.tref_out = self.tref_in[
+                np.logical_and(self.tref_in >= min(self.t),
+                               self.tref_in <= max(self.t))]
+            out_len = len(self.tref_out)
+            if out_len == 0:
+                # check that tref_in is in the allowed range
+                self.check_input_limits(
+                    self.tref_in, min(self.t), max(self.t))
         else:
-            return_dict.update({"tref_out": self.tref_out})
+            # Since we don't have the maximum and minimum allowed reference
+            # frequencies computed from the frequencies at the pericenters and
+            # apocenters, we simply set the maximum and minimum values to be
+            # the maximum and minimum of instantaneous f22, respectively.
+            f22_min = min(self.omega22) / (2 * np.pi)
+            f22_max = max(self.omega22) / (2 * np.pi)
+            self.fref_out = self.fref_in[
+                np.logical_and(self.fref_in >= f22_min,
+                               self.fref_in <= f22_max)]
+            out_len = len(self.fref_out)
+            if out_len == 0:
+                # check that fref_in is in the allowed range
+                self.check_input_limits(
+                    self.fref_in, f22_min, f22_max)
+        self.eccentricity = np.zeros(out_len)
+        self.mean_anomaly = np.zeros(out_len)
+        return self.make_return_dict_for_eccentricity_and_mean_anomaly()
+
+    def make_return_dict_for_eccentricity_and_mean_anomaly(self):
+        """Prepare a dictionary with reference time/freq, ecc, and mean anomaly.
+
+        In this function, we prepare a dictionary containing the measured
+        eccentricity, mean anomaly, and the reference time or frequency where
+        these are measured.
+
+        We also make sure that if the input reference time/frequency is scalar,
+        then the returned eccentricity and mean anomaly are also scalars. To do
+        this, we use the information about tref_in/fref_in that is provided by
+        the user. At the top of the measure_ecc function, we set ref_ndim to
+        identify whether the original input was scalar or array-like and use
+        that here.
+        """
+        # If the original input was scalar, convert the measured eccentricity,
+        # mean anomaly, etc., to scalar.
+        if self.ref_ndim == 0:
+            self.eccentricity = self.eccentricity[0]
```

Comment:
Shouldn't we double check that the size of the original array was 1 before converting to scalar?

Reply:
Done